examples/gaussian_process/plot_gpr_prior_posterior.py

```diff
@@ -3,76 +3,259 @@
-
-    # Plot prior
-    plt.figure(figsize=(8, 8))
-    plt.subplot(2, 1, 1)
-    X_ = np.linspace(0, 5, 100)
-    y_mean, y_std = gp.predict(X_[:, np.newaxis], return_std=True)
-    plt.plot(X_, y_mean, 'k', lw=3, zorder=9)
-    plt.fill_between(X_, y_mean - y_std, y_mean + y_std,
-                     alpha=0.2, color='k')
-    y_samples = gp.sample_y(X_[:, np.newaxis], 10)
-    plt.plot(X_, y_samples, lw=1)
-    plt.xlim(0, 5)
-    plt.ylim(-3, 3)
-    plt.title("Prior (kernel:  %s)" % kernel, fontsize=12)
-
-    # Generate data and fit GP
-    rng = np.random.RandomState(4)
-    X = rng.uniform(0, 5, 10)[:, np.newaxis]
-    y = np.sin((X[:, 0] - 2.5) ** 2)
-    gp.fit(X, y)
-
-    # Plot posterior
-    plt.subplot(2, 1, 2)
-    X_ = np.linspace(0, 5, 100)
-    y_mean, y_std = gp.predict(X_[:, np.newaxis], return_std=True)
-    plt.plot(X_, y_mean, 'k', lw=3, zorder=9)
-    plt.fill_between(X_, y_mean - y_std, y_mean + y_std,
-                     alpha=0.2, color='k')
-
-    y_samples = gp.sample_y(X_[:, np.newaxis], 10)
-    plt.plot(X_, y_samples, lw=1)
-    plt.scatter(X[:, 0], y, c='r', s=50, zorder=10, edgecolors=(0, 0, 0))
-    plt.xlim(0, 5)
-    plt.ylim(-3, 3)
-    plt.title("Posterior (kernel: %s)\n Log-Likelihood: %.3f"
-              % (gp.kernel_, gp.log_marginal_likelihood(gp.kernel_.theta)),
-              fontsize=12)
-    plt.tight_layout()
-
-plt.show()
+from sklearn.gaussian_process.kernels import RBF
+
+kernel = 1.0 * RBF(length_scale=1.0, length_scale_bounds=(1e-1, 10.0))
+gpr = GaussianProcessRegressor(kernel=kernel, random_state=0)
+
+fig, axs = plt.subplots(nrows=2, sharex=True, sharey=True, figsize=(10, 8))
+
+# plot prior
+plot_gpr_samples(gpr, n_samples=n_samples, ax=axs[0])
+axs[0].set_title("Samples from prior distribution")
+
+# plot posterior
+gpr.fit(X_train, y_train)
+plot_gpr_samples(gpr, n_samples=n_samples, ax=axs[1])
+axs[1].scatter(X_train[:, 0], y_train, color="red", zorder=10, label="Observations")
+axs[1].legend(bbox_to_anchor=(1.05, 1.5), loc="upper left")
+axs[1].set_title("Samples from posterior distribution")
+
+fig.suptitle("Radial Basis Function kernel", fontsize=18)
+plt.tight_layout()
+
+# %%
+print(f"Kernel parameters before fit:\n{kernel})")
+print(
+    f"Kernel parameters after fit: \n{gpr.kernel_} \n"
+    f"Log-likelihood: {gpr.log_marginal_likelihood(gpr.kernel_.theta):.3f}"
+)
+
+# %%
+# Rational Quadradtic kernel
+# ..........................
+from sklearn.gaussian_process.kernels import RationalQuadratic
+
+kernel = 1.0 * RationalQuadratic(length_scale=1.0, alpha=0.1, alpha_bounds=(1e-5, 1e10))
+gpr = GaussianProcessRegressor(kernel=kernel, random_state=0)
+
+fig, axs = plt.subplots(nrows=2, sharex=True, sharey=True, figsize=(10, 8))
+
+# plot prior
+plot_gpr_samples(gpr, n_samples=n_samples, ax=axs[0])
+axs[0].set_title("Samples from prior distribution")
+
+# plot posterior
+gpr.fit(X_train, y_train)
+plot_gpr_samples(gpr, n_samples=n_samples, ax=axs[1])
```

Comment:
This gives the following warnings:
```
sklearn/gaussian_process/_gpr.py:408: UserWarning: Predicted variances smaller than 0. Setting those variances to 0.
sklearn/gaussian_process/_gpr.py:449: RuntimeWarning: covariance is not positive-semidefinite.
```

I don't know enough about the implementation yet to remove it.

Reply:
I assume that the change of the `alpha_bounds` also solved this issue.